lib/bots/versionbot.ts

```diff
@@ -1550,6 +1571,157 @@ export class VersionBot extends ProcBot {
+					'in configuration');
+			}
+			if (tag.occurrence) {
+				if ((typeof tag.occurrence !== 'number') && (tag.occurrence !== 'all')) {
+					throw new Error(`Invalid occurrence value found for ${tag.name} definition`);
+				}
+			}
+			// Always use a lowered version of the tag.
+			tagCounts[tag.name.toLowerCase()] = 0;
+
+			if (tag.name === 'change-type') {
+				changeTypeFound = true;
+			}
+		}
+
+		// Always add the 'change-type' tag.
+		if (!changeTypeFound) {
+			tagDefinitions.push({
+				name: 'change-type',
+				values: '/\s*(patch|minor|major)\s*$/i'
+			});
+			tagCounts['change-type'] = 0;
+		}
+
+		// Go through each commit. For each tag, we determine if it's present or not
+		// and if it matches required values.
+		for (let commit of commits) {
+			const commitMessage: string = commit.commit.message;
+
+			// Split the commits up into lines, and find the last line with any whitespace in.
+			// Whilst we tend to ask for:
+			//  <header>
+			//
+			//  <body>
+			//
+			//  <footer>
+			// This code will actually let you get away with:
+			//  <header>
+			//
+			//  <footer>
+			// As sometimes a development patch may be self-explanatory in the header alone.
+			const lines = commitMessage.split('\n');
+			const lastLine = _.findLastIndex(lines, (line) => line.match(/^\s*$/) );
+
+			// If there's no match, then at the very least there's no footer, and the commit
+			// is in the wrong format (as there's no text to use in the logs).
+			if (lastLine > 0) {
+				// We should have a line index to join from, now.
+				lines.splice(0, lastLine);
+				const footer = lines.join('\n');
+
+				// For each tag, interrogate the footer and determine if the tag is
+				// present.
+				// We check for a valid instance of a tag. If there are duplicates,
+				// potentially with invalid values, we don't flag it. This is primarily
+				// as we don't know how scripts requiring these tags deal with them
+				// and have to assume they'll pick a valid tag out.
+				_.each(tagDefinitions, (tag) => {
+					// RE for the key. We want to capture the entire line
+					const keyRE = new RegExp(`^${tag.name}:(.*)$`, 'gmi');
+
+					// We need to compile the RE for the value, if it doesn't already exist.
+					// Strip the RE into a body and flags.
+					let valueRE: RegExp = /.*/;
+					if (tag.values) {
+						// Get flag values, we ignore everything apart from 'i'
+						const regExp = _.split(tag.values, '/');
+						const body = regExp[1];
+						const flags = regExp[2] || '';
+						if (!body) {
+							throw new Error(`Invalid RegExp found for ${tag.name} values`);
+						}
+
+						valueRE = new RegExp(body, flags);
+					}
+
+					// Try and match the key.
+					// We want all matches.
+					let valueMatches = [];
+					let match = keyRE.exec(footer);
+					while (match) {
+						valueMatches.push(match[1]);
+						match = keyRE.exec(footer);
+					}
```

Comment:
@hedss it's per-project, you can implement it here and not owe the world any explanation :)
You can check [how I did it](https://github.com/resin-io/resin-vpn/issues/25) on the vpn. Note that it has 2 PRs because it started as `precommit` and switched to `prepush` after some complaints. You can keep it on `precommit` if you want, specially if it's only for linting and doesn't even run tests.